app/javascript/controllers/tiptap_controller.ts

```diff
@@ -0,0 +1,567 @@
+  }
+
+  private get menuButtons() {
+    return this.buttonTargets.map(
+      (menuButton) => menuButton.dataset.tiptapAction as string
+    );
+  }
+}
+
+function createEditor({
+  editorElement,
+  content,
+  tags,
+  buttons,
+  onChange
+}: {
+  editorElement: Element;
+  content?: JSONContent;
+  tags: TagSchema[];
+  buttons: string[];
+  onChange(change: { editor: Editor }): void;
+}): Editor {
+  const options = getEditorOptions(editorElement, tags, buttons, content);
+  const editor = new Editor(options);
+  editor.on('transaction', onChange);
+  return editor;
+}
+
+function getEditorOptions(
+  element: Element,
+  tags: TagSchema[],
+  actions: string[],
+  content?: JSONContent
+): Partial<EditorOptions> {
+  const extensions: Extensions = [];
+  for (const action of actions) {
+    switch (action) {
+      case 'bold':
+        extensions.push(Bold);
+        break;
+      case 'italic':
+        extensions.push(Italic);
+        break;
+      case 'underline':
+        extensions.push(Underline);
+        break;
+      case 'strike':
+        extensions.push(Strike);
+        break;
+      case 'highlight':
+        extensions.push(Highlight);
+        break;
+      case 'bulletList':
+        extensions.push(BulletList);
+        break;
+      case 'orderedList':
+        extensions.push(OrderedList);
+        break;
+      case 'left':
+      case 'center':
+      case 'right':
+      case 'justify':
+        extensions.push(
+          TextAlign.configure({
+            types: actions.includes('title')
+              ? ['headerColumn', 'title', 'footer', 'heading', 'paragraph']
+              : ['heading', 'paragraph']
+          })
+        );
+        break;
+      case 'title':
+        extensions.push(Header, HeaderColumn, Title, Footer);
+        break;
+      case 'heading2':
+      case 'heading3':
+        extensions.push(Heading.configure({ levels: [2, 3] }));
+        break;
+    }
+  }
+
+  if (actions.includes('bulletList') || actions.includes('orderedList')) {
+    extensions.push(ListItem);
+  }
+  if (tags.length > 0) {
+    extensions.push(
+      Mention.configure({
+        renderLabel({ node }) {
+          return `--${node.attrs.label}--`;
+        },
+        HTMLAttributes: {
+          class: 'fr-badge fr-badge--sm fr-badge--info fr-badge--no-icon'
+        },
+        suggestion: createSuggestionMenu(tags, element)
+      })
+    );
+  }
+
+  return {
+    element,
+    content,
+    editorProps: { attributes: { class: 'fr-input' } },
+    extensions: [
+      actions.includes('title') ? DocumentWithHeader : Document,
+      Hystory,
+      Typography,
+      Gapcursor,
+      Paragraph,
+      Text,
+      ...extensions
+    ]
+  };
+}
+
+type EditorAction = {
+  run(): void;
+  isActive(): boolean;
+  isDisabled(): boolean;
+};
+
+function getAction(editor: Editor, button: HTMLButtonElement): EditorAction {
+  return tiptapActionSchema.parse(button.dataset)(editor);
+}
```

Comment:
alors, est ce que c'est pas une façon très compliquée de faire

```suggestion
function getAction(editor: Editor, button: HTMLButtonElement): EditorAction {
  return EDITOR_ACTIONS[button.le_path_vers_le_dataset_qui_va_bien](editor);
}
```